<a href="https://colab.research.google.com/github/GaelNgayou/Projet-Data/blob/main/Cas_Covid_Kaggle.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [45]:
import pandas as pd
from scipy.interpolate import CubicSpline
import numpy as np

In [46]:
df = pd.read_csv('Covid Live.csv')

In [47]:
df

# Country,\nOther Total\nCases Total\nDeaths  New\nDeaths  \
0      1             USA   98,166,904     1,084,282          NaN   
1      2           India   44,587,307       528,629          NaN   
2      3          France   35,342,950       155,078          NaN   
3      4          Brazil   34,706,757       686,027          NaN   
4      5         Germany   33,312,373       149,948          NaN   
..   ...             ...          ...           ...          ...   
225  226            Niue           80           NaN          NaN   
226  227    Vatican City           29           NaN          NaN   
227  228          Tuvalu           20           NaN          NaN   
228  229  Western Sahara           10             1          NaN   
229  230      MS Zaandam            9             2          NaN   

    Total\nRecovered Active\nCases Serious,\nCritical Tot Cases/\n1M pop  \
0         94,962,112     2,120,510              2,970            293,206   
1         44,019,095        39,583                698             31,698   
2         34,527,115       660,757                869            538,892   
3         33,838,636       182,094              8,318            161,162   
4         32,315,200       847,225              1,406            397,126   
..               ...           ...                ...                ...   
225               80             0                NaN             49,322   
226               29             0                NaN             36,295   
227              NaN            20                NaN              1,658   
228                9             0                NaN                 16   
229                7             0                NaN                NaN   

    Deaths/\n1M pop   Total\nTests Tests/\n1M pop     Population  
0             3,239  1,118,158,870      3,339,729    334,805,269  
1               376    894,416,853        635,857  1,406,631,776  
2             2,365    271,490,188      4,139,547     65,584,518  
3             3,186     63,776,166        296,146    215,353,593  
4             1,788    122,332,384      1,458,359     83,883,596  
..              ...            ...            ...            ...  
225             NaN            NaN            NaN          1,622  
226             NaN            NaN            NaN            799  
227             NaN            NaN            NaN         12,066  
228               2            NaN            NaN        626,161  
229             NaN            NaN            NaN            NaN  

[230 rows x 13 columns]

In [48]:
df.shape

(230, 13)

In [49]:
df.dtypes

#                       int64
Country,\nOther        object
Total\nCases           object
Total\nDeaths          object
New\nDeaths           float64
Total\nRecovered       object
Active\nCases          object
Serious,\nCritical     object
Tot Cases/\n1M pop     object
Deaths/\n1M pop        object
Total\nTests           object
Tests/\n1M pop         object
Population             object
dtype: object

In [50]:
colonnes_converties = [colonne for colonne in df.columns if colonne != ['Country,\nOther','New\nDeaths','#']]

In [51]:
df[colonnes_converties] = df[colonnes_converties].replace({',': ''}, regex=True)

In [52]:
nbre_nan = df.isna().sum()

In [53]:
nbre_nan

#                       0
Country,\nOther         0
Total\nCases            0
Total\nDeaths           5
New\nDeaths           227
Total\nRecovered       16
Active\nCases          15
Serious,\nCritical     83
Tot Cases/\n1M pop      2
Deaths/\n1M pop         7
Total\nTests           16
Tests/\n1M pop         16
Population              2
dtype: int64

In [54]:
interpolation = ['Total\nDeaths', 'New\nDeaths', 'Total\nRecovered', 'Active\nCases', 'Serious,\nCritical', 'Tot Cases/\n1M pop', 'Deaths/\n1M pop', 'Total\nTests', 'Tests/\n1M pop', 'Population']

In [55]:
df[interpolation] = df[interpolation].interpolate()

In [56]:
df[interpolation]

Total\nDeaths  New\nDeaths Total\nRecovered Active\nCases  \
0         1084282          NaN         94962112       2120510   
1          528629          NaN         44019095         39583   
2          155078          NaN         34527115        660757   
3          686027          NaN         33838636        182094   
4          149948          NaN         32315200        847225   
..            ...          ...              ...           ...   
225           NaN          3.0               80             0   
226           NaN          3.0               29             0   
227           NaN          3.0              NaN            20   
228             1          3.0                9             0   
229             2          3.0                7             0   

    Serious,\nCritical Tot Cases/\n1M pop Deaths/\n1M pop Total\nTests  \
0                 2970             293206            3239   1118158870   
1                  698              31698             376    894416853   
2                  869             538892            2365    271490188   
3                 8318             161162            3186     63776166   
4                 1406             397126            1788    122332384   
..                 ...                ...             ...          ...   
225                NaN              49322             NaN          NaN   
226                NaN              36295             NaN          NaN   
227                NaN               1658             NaN          NaN   
228                NaN                 16               2          NaN   
229                NaN                NaN             NaN          NaN   

    Tests/\n1M pop  Population  
0          3339729   334805269  
1           635857  1406631776  
2          4139547    65584518  
3           296146   215353593  
4          1458359    83883596  
..             ...         ...  
225            NaN        1622  
226            NaN         799  
227            NaN       12066  
228            NaN      626161  
229            NaN         NaN  

[230 rows x 10 columns]

In [57]:
print(df[interpolation].isna().any())

Total\nDeaths         True
New\nDeaths           True
Total\nRecovered      True
Active\nCases         True
Serious,\nCritical    True
Tot Cases/\n1M pop    True
Deaths/\n1M pop       True
Total\nTests          True
Tests/\n1M pop        True
Population            True
dtype: bool


In [58]:
(df[interpolation].isna().mean().mean() * 100).round(2)

7.26

In [59]:
df[interpolation]

Total\nDeaths  New\nDeaths Total\nRecovered Active\nCases  \
0         1084282          NaN         94962112       2120510   
1          528629          NaN         44019095         39583   
2          155078          NaN         34527115        660757   
3          686027          NaN         33838636        182094   
4          149948          NaN         32315200        847225   
..            ...          ...              ...           ...   
225           NaN          3.0               80             0   
226           NaN          3.0               29             0   
227           NaN          3.0              NaN            20   
228             1          3.0                9             0   
229             2          3.0                7             0   

    Serious,\nCritical Tot Cases/\n1M pop Deaths/\n1M pop Total\nTests  \
0                 2970             293206            3239   1118158870   
1                  698              31698             376    894416853   
2                  869             538892            2365    271490188   
3                 8318             161162            3186     63776166   
4                 1406             397126            1788    122332384   
..                 ...                ...             ...          ...   
225                NaN              49322             NaN          NaN   
226                NaN              36295             NaN          NaN   
227                NaN               1658             NaN          NaN   
228                NaN                 16               2          NaN   
229                NaN                NaN             NaN          NaN   

    Tests/\n1M pop  Population  
0          3339729   334805269  
1           635857  1406631776  
2          4139547    65584518  
3           296146   215353593  
4          1458359    83883596  
..             ...         ...  
225            NaN        1622  
226            NaN         799  
227            NaN       12066  
228            NaN      626161  
229            NaN         NaN  

[230 rows x 10 columns]

Etant donné que nous avons 7.26% de valeurs aberrantes dans le df et après avoir essayé plusieurs méthodes d'interpolation sans succès, j'ai décidé de remplacer les valeurs Nan par 0 et d'isoler tous les 0 lors de l'analyse que j'effectuerai sur PowerBI. Je testerai les méthodes d'interpolation de PowerBI plus tard.

In [60]:
df[interpolation] = df[interpolation].fillna(0)

In [61]:
df[interpolation]

Total\nDeaths  New\nDeaths Total\nRecovered Active\nCases  \
0         1084282          0.0         94962112       2120510   
1          528629          0.0         44019095         39583   
2          155078          0.0         34527115        660757   
3          686027          0.0         33838636        182094   
4          149948          0.0         32315200        847225   
..            ...          ...              ...           ...   
225             0          3.0               80             0   
226             0          3.0               29             0   
227             0          3.0                0            20   
228             1          3.0                9             0   
229             2          3.0                7             0   

    Serious,\nCritical Tot Cases/\n1M pop Deaths/\n1M pop Total\nTests  \
0                 2970             293206            3239   1118158870   
1                  698              31698             376    894416853   
2                  869             538892            2365    271490188   
3                 8318             161162            3186     63776166   
4                 1406             397126            1788    122332384   
..                 ...                ...             ...          ...   
225                  0              49322               0            0   
226                  0              36295               0            0   
227                  0               1658               0            0   
228                  0                 16               2            0   
229                  0                  0               0            0   

    Tests/\n1M pop  Population  
0          3339729   334805269  
1           635857  1406631776  
2          4139547    65584518  
3           296146   215353593  
4          1458359    83883596  
..             ...         ...  
225              0        1622  
226              0         799  
227              0       12066  
228              0      626161  
229              0           0  

[230 rows x 10 columns]

In [62]:
df[interpolation].dtypes

Total\nDeaths          object
New\nDeaths           float64
Total\nRecovered       object
Active\nCases          object
Serious,\nCritical     object
Tot Cases/\n1M pop     object
Deaths/\n1M pop        object
Total\nTests           object
Tests/\n1M pop         object
Population             object
dtype: object

In [63]:
df[interpolation] = df[interpolation].astype('int64')

In [64]:
df[interpolation]

Total\nDeaths  New\nDeaths  Total\nRecovered  Active\nCases  \
0          1084282            0          94962112        2120510   
1           528629            0          44019095          39583   
2           155078            0          34527115         660757   
3           686027            0          33838636         182094   
4           149948            0          32315200         847225   
..             ...          ...               ...            ...   
225              0            3                80              0   
226              0            3                29              0   
227              0            3                 0             20   
228              1            3                 9              0   
229              2            3                 7              0   

     Serious,\nCritical  Tot Cases/\n1M pop  Deaths/\n1M pop  Total\nTests  \
0                  2970              293206             3239    1118158870   
1                   698               31698              376     894416853   
2                   869              538892             2365     271490188   
3                  8318              161162             3186      63776166   
4                  1406              397126             1788     122332384   
..                  ...                 ...              ...           ...   
225                   0               49322                0             0   
226                   0               36295                0             0   
227                   0                1658                0             0   
228                   0                  16                2             0   
229                   0                   0                0             0   

     Tests/\n1M pop  Population  
0           3339729   334805269  
1            635857  1406631776  
2           4139547    65584518  
3            296146   215353593  
4           1458359    83883596  
..              ...         ...  
225               0        1622  
226               0         799  
227               0       12066  
228               0      626161  
229               0           0  

[230 rows x 10 columns]

In [71]:
colonnes_concat = ['#','Country,\nOther','New\nDeaths']

In [72]:
df_concat = df[colonnes_concat]

In [73]:
df2 = df[interpolation]

In [74]:
df_final = pd.concat([df_concat, df2], axis=1)

In [75]:
df_final

# Country,\nOther  New\nDeaths  Total\nDeaths  New\nDeaths  \
0      1             USA            0        1084282            0   
1      2           India            0         528629            0   
2      3          France            0         155078            0   
3      4          Brazil            0         686027            0   
4      5         Germany            0         149948            0   
..   ...             ...          ...            ...          ...   
225  226            Niue            3              0            3   
226  227    Vatican City            3              0            3   
227  228          Tuvalu            3              0            3   
228  229  Western Sahara            3              1            3   
229  230      MS Zaandam            3              2            3   

     Total\nRecovered  Active\nCases  Serious,\nCritical  Tot Cases/\n1M pop  \
0            94962112        2120510                2970              293206   
1            44019095          39583                 698               31698   
2            34527115         660757                 869              538892   
3            33838636         182094                8318              161162   
4            32315200         847225                1406              397126   
..                ...            ...                 ...                 ...   
225                80              0                   0               49322   
226                29              0                   0               36295   
227                 0             20                   0                1658   
228                 9              0                   0                  16   
229                 7              0                   0                   0   

     Deaths/\n1M pop  Total\nTests  Tests/\n1M pop  Population  
0               3239    1118158870         3339729   334805269  
1                376     894416853          635857  1406631776  
2               2365     271490188         4139547    65584518  
3               3186      63776166          296146   215353593  
4               1788     122332384         1458359    83883596  
..               ...           ...             ...         ...  
225                0             0               0        1622  
226                0             0               0         799  
227                0             0               0       12066  
228                2             0               0      626161  
229                0             0               0           0  

[230 rows x 13 columns]

In [76]:
df_final.isna().sum()

#                     0
Country,\nOther       0
New\nDeaths           0
Total\nDeaths         0
New\nDeaths           0
Total\nRecovered      0
Active\nCases         0
Serious,\nCritical    0
Tot Cases/\n1M pop    0
Deaths/\n1M pop       0
Total\nTests          0
Tests/\n1M pop        0
Population            0
dtype: int64

In [77]:
df_final.dtypes

#                      int64
Country,\nOther       object
New\nDeaths            int64
Total\nDeaths          int64
New\nDeaths            int64
Total\nRecovered       int64
Active\nCases          int64
Serious,\nCritical     int64
Tot Cases/\n1M pop     int64
Deaths/\n1M pop        int64
Total\nTests           int64
Tests/\n1M pop         int64
Population             int64
dtype: object

In [82]:
df_final.to_csv('Nvo_Covid_Live.csv', index=False)